sdk/core/azure-core-http-httpurlconnection/src/main/java/com/azure/core/http/httpurlconnection/HttpUrlConnectionAsyncClient.java

```diff
@@ -0,0 +1,206 @@
+package com.azure.core.http.httpurlconnection;
+
+import com.azure.core.http.*;
+import com.azure.core.http.httpurlconnection.implementation.HttpUrlConnectionResponse;
+import com.azure.core.util.*;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import com.azure.core.http.HttpHeader;
+
+import java.io.*;
+import java.net.*;
+import java.nio.ByteBuffer;
+import java.time.Duration;
+import java.util.*;
+
+/**
+ * This class provides a HttpUrlConnection implementation for the {@link HttpClient} interface.
+ * Creating an instance of this class can be achieved by using the {@link HttpUrlConnectionAsyncClientBuilder} class.
+ *
+ * @see HttpClient
+ * @see HttpUrlConnectionAsyncClientBuilder
+ */
+public class HttpUrlConnectionAsyncClient implements HttpClient {
+    private static final ClientLogger LOGGER = new ClientLogger(HttpUrlConnectionAsyncClient.class);
+    private final Duration connectionTimeout;
+    private final ProxyOptions proxyOptions;
+    private final Configuration configuration;
+
+    HttpUrlConnectionAsyncClient(Duration connectionTimeout, ProxyOptions proxyOptions, Configuration configuration) {
+        this.connectionTimeout = connectionTimeout;
+        this.proxyOptions = proxyOptions;
+        this.configuration = configuration;
+    }
+
+    @Override
+    public Mono<HttpResponse> send(HttpRequest httpRequest) {
+        return sendAsync(httpRequest, Context.NONE);
+    }
+
+    @Override
+    public Mono<HttpResponse> send(HttpRequest request, Context context) {
+        return sendAsync(request, context);
+    }
+
+    @Override
+    public HttpResponse sendSync(HttpRequest httpRequest, Context context) {
+        return sendAsync(httpRequest, context).block();
```

Comment:
Since `HttpURLConnection` is inherently synchronous we should create a separate implementation for this API instead of blocking the asynchronous API. But to help code reuse we can create helper methods that can be shared across both the asynchronous and synchronous API.

Reply:
@alzimmermsft are you suggesting a `HttpUrlConnectionSyncClient` class, or to rewrite the `sendSync` method so it's not *just* blocking the async method?

The only method (of `connect`, `sendRequest`, `receiveResponse`) that's also async is `sendRequest` once we successfully un`.block()` writing the body.

Reply:
I would suggest the latter of rewriting `sendSync` to not just be `sendAsync.block`.